### PR TITLE
Update AMO submission process docs

### PR DESCRIPTION
**Pull request by dotproto**
This PR updates the "Listing on AMO" docs to reflect the current user flow seen on addons.mozilla.org.

Note that this PR does not update the header for `submitting-an-add-on.md` the same values will be used in #1704.

### CODE REVIEW

File: src/content/documentation/publish/submitting-an-add-on.md [hunk 1]
```diff
@@ -75,23 +75,34 @@ Choose:
...
-6. Select the add-on's compatible platform(s).
-7. Click **Continue**.
-8. The Describe Add-on page displays. <br/> ![Submit an add-on page](/assets/img/publish/Submit_add_on_describe_add_on.png) <br/>
+5. Select the add-on's compatible platform(s).
+
+6. Click **Continue**.
+
+7. Select whether or not you need to provide your source code. <br/><br/> <img src="/assets/img/publish/Submit_add_on_source_code.png" style="box-shadow:0px 0px 0.5em gray;" />
+
+    If you do, select **Yes** and a new section will appear that describes what you will need to submit. Click **Browse** and locate and upload your source code package. See [Source code submission](/documentation/publish/source-code-submission/) for more information.
```
Comment (rebloor) [inline src/content/documentation/publish/submitting-an-add-on.md:84]:
```suggestion
    If you select **Yes**, a section displays that describes what you need to submit. Click **Browse** and locate and upload your source code package. See [Source code submission](/documentation/publish/source-code-submission/) for more information.
```

File: src/content/documentation/publish/submitting-an-add-on.md [hunk 2]
```diff
@@ -75,23 +75,34 @@ Choose:
...
+    
+    If you do not, select **No**.
+
```
Comment (rebloor) [inline src/content/documentation/publish/submitting-an-add-on.md:87]:
```suggestion
```

File: src/content/documentation/publish/submitting-an-add-on.md [hunk 3]
```diff
@@ -75,23 +75,34 @@ Choose:
...
-   - Select up to 2 Firefox categories for this add-on: select categories that describe your add-on.
-   - Select up to 2 Firefox for Android categories for this add-on: select categories that describe your add-on.
-   - Support email and website: an email address and website where users can get in touch when they have questions, issues, or compliments.
-   - License: select the appropriate license for your add-on. Click Details to learn more about each license.
-   - This add-on has a privacy policy: if any data is being transmitted from the user’s device, a privacy policy is required explaining what is being sent and how it’s used. Check this box and provide the privacy policy.
-   - Notes for Reviewers: information that will assist the AMO reviewer, such as log in details for a dummy account, source code information, or similar.
+   - **Name:** your add-on's name.
+   - **Add-on URL:** the URL for your add-on on AMO. A URL will be automatically assigned based on your add-on's name, to change this, click Edit. Note: The URL must be unique, you'll be warned if another add-on is using your chosen URL and you'll have to enter a different one before you can set the URL.
```
Comment (rebloor) [inline src/content/documentation/publish/submitting-an-add-on.md:92]:
```suggestion
   - **Add-on URL:** the URL for your add-on on AMO. A URL will be automatically assigned based on your add-on's name, to change this, click **Edit**. Note: The URL must be unique, you'll be warned if another add-on is using your chosen URL and you'll have to enter a different one before you can set the URL.
```

File: src/content/documentation/publish/submitting-an-add-on.md [hunk 4]
```diff
@@ -75,23 +75,34 @@ Choose:
...
+   - **Summary:** a useful and descriptive short summary of your add-on.
+   - **Description:** a longer description that provides users with a more detailed oveview of the extensin's features and functionality.
+   - **This add-on is experimental:** your add-on is experimental or otherwise not ready for general use. The add-on will be listed but with reduced visibility. You can remove this flag when your add-on is ready for general use.
+   - **This add-on requires payment, non-free services or software, or additional hardware:** indicate if your add-on requires users to make an additional purchase for it to work fully.
+   - **Select up to 2 Firefox categories for this add-on:** select categories that describe your add-on.
+   - **Select up to 2 Firefox for Android categories for this add-on:** select categories that describe your add-on.
+   - **Support email** and **Support website:** an email address and website where users can get in touch when they have questions, issues, or compliments.
+   - **License:** select the appropriate license for your add-on. Click Details to learn more about each license.
```
Comment (rebloor) [inline src/content/documentation/publish/submitting-an-add-on.md:100]:
```suggestion
   - **License:** select the appropriate license for your add-on. Click **Details** to learn more about each license.
```

**Review by rebloor — COMMENTED** [review]
I've left some suggestions

**Comment by dotproto** [comment]
Thanks for apply the suggested changes and making additional updates, @rebloor! I just added a little more polish. I think this is good to go.

**Review by rebloor — APPROVED** [review]
@dotproto sorry for the delay. I'm not sure this page should have moved from markdown for the images. I'll look at that later. I've added alt text to all the images.

**Pull request merged**